java/code/src/com/redhat/rhn/taskomatic/task/PackageCleanup.java

```diff
@@ -74,13 +77,29 @@ else if (log.isDebugEnabled()) {
 
             // Reset the queue (table)
             resetQueue();
+
+            // Change org for orphan vendor packages that a user can delete them
+            changeOrgForOrphanVendorPackages();
         }
         catch (Exception e) {
             log.error(e.getMessage(), e);
             throw new JobExecutionException(e);
         }
     }
 
+    private void changeOrgForOrphanVendorPackages() {
+        LocalDateTime now = LocalDateTime.now();
+        if (now.getHour() == 21 && now.getMinute() >= 49) {
```

Comment:
This feels strange to me; maybe consider making this at least a variable or a helper method instead explaining in a comment? I guess looking at this, I don't understand why we're executing this only at a specified hour after a specified minute instead of every time the method is called.

Reply:
This job run too often. It run every 10 minutes. We want this cleanup only 1 time per day to prevent the problem we have. If we would run it every time, we would just move the problem from python to java code.

Minute 49 is just to be sure that when taskomatic schedule it not at 50, but 51, the job is not skipped.
At the next run we already have hour "22" which would not match. So we can be sure that it only run 1 time per day.

Reply:
Maybe we could add it to the method comment? 

Ideally, we'd remove random constant numbers (like `hour == 21` && minute >= 49` -> `isDailyExecutionWindow()` though that still leaves the `21` and `49` constants being just that, random constants deep in a class) but probably a comment would be good enough since I'm not sure how to surface the `21` and `49` constants to a more readable place.

Reply:
This has changed as well now. Check the new code